src/gui/control_panel.rs

```diff
@@ -1,21 +1,35 @@
+use std::collections::HashMap;
+use std::mem::replace;
+
 use eframe::egui;
 use egui::{CtxRef, Ui};
 
-use crate::dialog;
+use crate::{
+    dialog,
+    event::{Event, EventGenerator},
+};
 
-pub enum ControlAction {
+#[repr(u8)]
```

Comment:
```
type EventType = u8;
#[repr(EventType)]
```
Would that work?

Reply:
I think is not possible, only primitive type name is allowed. But anymore required due to the casting as u8.